app/v1/controllers/validators/CreateAndAmendOtherDeductionsValidatorFactory.scala

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.validators
+
+import api.controllers.validators.Validator
+import api.controllers.validators.resolvers.{ResolveJsonObject, ResolveNino, ResolveParsedNumber, ResolveTaxYear}
+import api.models.errors._
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import cats.implicits._
+import play.api.libs.json.JsValue
+import v1.models.request.createAndAmendOtherDeductions.{CreateAndAmendOtherDeductionsBody, CreateAndAmendOtherDeductionsRequestData, Seafarers}
+
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import scala.util.{Failure, Success, Try}
+
+class CreateAndAmendOtherDeductionsValidatorFactory {
+
+  private val resolveJson = new ResolveJsonObject[CreateAndAmendOtherDeductionsBody]()
+
+  def validator(nino: String, taxYear: String, body: JsValue): Validator[CreateAndAmendOtherDeductionsRequestData] =
+    new Validator[CreateAndAmendOtherDeductionsRequestData] {
+
+      def validate: Validated[Seq[MtdError], CreateAndAmendOtherDeductionsRequestData] =
+        (
+          ResolveNino(nino),
+          ResolveTaxYear(taxYear),
+          resolveJson(body)
+        ).mapN(CreateAndAmendOtherDeductionsRequestData) andThen validateBody
+
+      private def validateBody(
+          parsed: CreateAndAmendOtherDeductionsRequestData): Validated[Seq[MtdError], CreateAndAmendOtherDeductionsRequestData] = {
+        import parsed.body._
+        List(
+          validateBodyFieldFormat(seafarers),
+          validateBodyDateRange(seafarers)
+        ).traverse(identity)
+          .map(_ => parsed)
+      }
+
+    }
+
+  private def validateBodyFieldFormat(seafarers: Option[Seq[Seafarers]]): Validated[Seq[MtdError], Unit] = {
+    val validationResult = seafarers
+      .getOrElse(Seq.empty[Seafarers])
+      .zipWithIndex
+      .traverse_ { case (item, index) => validateSeafarers(item, index) }
+
+    validationResult
+  }
+
+  private def validateBodyDateRange(seafarers: Option[Seq[Seafarers]]): Validated[Seq[MtdError], Unit] = {
+    val validationResult = seafarers
+      .getOrElse(Seq.empty[Seafarers])
```

Comment:
Can just be "getOrElse(Nil)"
But rather than have the getOrElse at the start, could instead do:
```
seafarers
      .map(_.zipWithIndex.traverse_ { case (item, index) =>
        validateToDateBeforeFromDate(item, index)
      })
      .getOrElse(Valid(()))
```

Reply:
(This probably wasn't your code originally, but maybe still worth tidying up as it might help clarify the validator/resolver conversion issues...)